messages/teleporter/message_manager.go

```diff
@@ -323,3 +294,40 @@ func (m *messageManager) parseTeleporterMessage(warpMessageID ids.ID, warpPayloa
 	}
 	return teleporterMessage, nil
 }
+
+func (m *messageManager) getTeleporterMessenger(destinationClient vms.DestinationClient) (*teleportermessenger.TeleporterMessenger, error) {
+	client, ok := destinationClient.Client().(ethclient.Client)
+	if !ok {
+		m.logger.Error(
+			"Destination client is not an Ethereum client.",
+			zap.String("destinationBlockchainID", destinationClient.DestinationBlockchainID().String()),
+		)
+		return nil, errors.New("destination client is not an Ethereum client")
+	}
+
+	// Get the teleporter messenger contract
+	teleporterMessenger, err := teleportermessenger.NewTeleporterMessenger(common.BytesToAddress(m.protocolAddress[:]), client)
```

Comment:
So the call above to `destinationClient.Client().(ethclient.Client)` seems to me like a pretty big footgun. What I think we should do instead, is on this line we call `NewTeleporterMessengerCaller` instead of `NewTeleporterMessenger`. This gives a read-only version of the abi-binding, which only requires the client to have implemented `CodeAt` and `CallContract`. I propose we add those two functions to the `DestinationClient` interface, and that way we don't need to do any casting.

Reply:
since the call `destinationClient.Client().(ethclient.Client)` was already here before, and we have a separate issue https://github.com/ava-labs/awm-relayer/issues/132 to address changing this call, don't think it's worth addressing in this PR, and I will defer it for the new issue.

Reply:
Yeah sure! Sorry, didn't notice this was already a part of the previous code.